core/src/smtp/mod.rs

```diff
@@ -62,6 +63,13 @@ pub async fn check_smtp(
 			.await
 			.map_err(|err| err.into());
 	}
+	if input.gmail_use_api
+		&& (host_lowercase.contains("gmail") || host_lowercase.contains("googlemail"))
```

Comment:
If using Google Apps/Workspace, this could actually be _any_ domain…? So should this just be `if input.gmail_use_api` with no further checks?

Reply:
`host` here refers to the MX host. I just tested:
- with `@gmail.com`, the hostnames look like `gmail-smtp-in.l.google.com.`
- with `@<company.com>` (where company uses custom domain with Google), the hostnames look like `*.aspmx.l.google.com.`

So I believe this should be good:
```suggestion
		&& host_lowercase.contains("gmail.com")
```